src/main/java/uk/gov/hmcts/cmc/claimstore/documents/DocumentUploader.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.cmc.claimstore.documents;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.documents.output.PDF;
+import uk.gov.hmcts.cmc.claimstore.events.DocumentGeneratedEvent;
+import uk.gov.hmcts.cmc.claimstore.services.ClaimService;
+import uk.gov.hmcts.cmc.claimstore.services.document.DocumentManagementService;
+
+@Component
+@ConditionalOnProperty(prefix = "feature_toggles", name = "document_management", havingValue = "true")
+public class DocumentUploader {
+
+    private final DocumentManagementService documentManagementService;
+    private final ClaimService claimService;
+
+    @Autowired
+    public DocumentUploader(final DocumentManagementService documentManagementService,
+                            final ClaimService claimService) {
+        this.documentManagementService = documentManagementService;
+        this.claimService = claimService;
+    }
+
+    @EventListener
+    public void uploadIntoDocumentManagementStore(DocumentGeneratedEvent event) {
+        event.getDocuments().forEach(document -> {
+            final String documentSelfPath = this.documentManagementService.uploadDocument(event.getAuthorisation(),
+                document.getFilename(), document.getBytes(), PDF.CONTENT_TYPE);
+
+            if (document.getFilename().endsWith("sealed-claim.pdf")) {
```

Comment:
shouldn't we be storing the URLs to all documents not just the sealed claim one?
This looks like a hack

Reply:
u r right, we uploading pin response letter now, but not reading from document management

Reply:
Agree @timja-kainos , it looks like a hack. I planned to put PIN letter into DM when T1 plays a story to completely integrate with DM (responses / CCJs / offers)